### ISSUE TITLE
nbgrader + ipywidgets causes timeouts

### ISSUE THREAD

**Issue by williamstein** [sagemathinc/cocalc#6137]
In comments on #5276 it was reported that combining ipywidgets with nbgrader causes timeouts/hangs/resets.  Fix this.

**Comment by mforbes** [comment]
Some magics also cause nbgrader to timeout:

```
import qiskit.tools.jupyter
%qiskit_version_table
```

One can install qiskit as follows:

```python
try:
    import qiskit
except ImportError:
    import sys
    !{sys.executable} -m pip install --user qiskit[visualization]
    print("Restart your kernel and try again")
```

**Comment by williamstein** [comment]
I wonder if I should add some sort of way to add metadata to the instructor notebook that has the impact: "do not include this cell in any way when grading?"

That seems easy to implement.  Doing something specific to ipywidgets or qiskit, etc., seems impossibly hard overall, since things like this will keep popping up.  Alternatively, changing the instructor notebook slightly could -- at any point, even after assigning and collecting -- change how grading happens.  And it would be easy to implement something to keep cells from being included in the grading process.  Thoughts?

**Comment by mforbes** [comment]
Excepting a way to fix the underlying issues, the real problem for me is that there is no easy way as an instructor to figure out which cells might be problematic.  Running **Validate** works fine, and it is only after collecting and grading assignments that the instructor finds out there is a problem.  Is there some way for "Validate" to be run properly through the nbgrader machinery? (Maybe create a "Test Student" account for doing this... of course with a unique ID that does not send poor Mr. Student to [Null Island](https://en.wikipedia.org/wiki/Null_Island)).

**Comment by williamstein** [comment]
> Excepting a way to fix the underlying issues,

That's a very good point.  I should just fix the underlying issue.  I wonder precisely how these things cause a hang.

**Comment by mforbes** [comment]
Is there a way to simulate how nbgrader is run?  I don't have a very good idea what it does under the hood.

**Comment by williamstein** [comment]
Good news -- I found what I think was the bug and fixed it in 5b0883f3639b26d556504697135b65141b90609a

Basically some code (especially widgets) accessed the "_actions" field of some class, which isn't set when using nbgrader.  

The fix should be live within an hour, if all the building/testing works fine.   The fix requires the project that will actually be running the code to be restarted if it is currently running.  Of course, any project that starts after the fix is deployed is updated to have the fix.
